avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -0,0 +1,291 @@
+package com.linkedin.avro.fastserde;
+
+import java.io.IOException;
+import java.util.AbstractList;
+import java.util.Collection;
+import java.util.Iterator;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.io.Decoder;
+
+
+/**
+ * This is a re-implementation of Avro's {@link GenericData.Array} class.
```

Comment:
Should we leave a `TODO: Provide arrays for the other primitive types`?

Reply:
Sure.